lib/sinatra/base.rb

```diff
@@ -1834,7 +1834,7 @@ def self.force_encoding(data, *) data end
 
     reset!
 
-    set :environment, (ENV['RACK_ENV'] || :development).to_sym
+    set :environment, (ENV['SINATRA_ENV'] || :development).to_sym
```

Comment:
I think enabling SINATRA_ENV, with a deprecation message when falling back to RACK_ENV, and then entirely disabling it in 2.0 makes sense.